integritybackend/numbers.py

```diff
@@ -35,65 +35,82 @@ def register(
-                ("assetSha256", asset_sha256),
-                ("assetMimetype", asset_mime_type),
-                ("assetTimestampCreated", asset_timestamp_created),
-                ("custom", json.dumps(custom)),
-            ]
-        else:
-            nft_metadata = {
-                "name": asset_name,
-                "description": asset_description,
-                "external_url": f"ipfs://{asset_cid}",
+            registration_data = {
+                "encodingFormat": asset_mime_type,
+                "assetCid": asset_cid,
+                "assetSha256": asset_sha256,
+                "assetTimestampCreated": asset_timestamp_created,
                 "custom": custom,
+                "testnet": True,  # TODO
+                "assetCreator": "Starling Lab",
             }
+        else:
+            # nft_metadata = {
+            #     "name": asset_name,
+            #     "description": asset_description,
+            #     "external_url": f"ipfs://{asset_cid}",
+            #     "custom": custom,
+            # }
+            #
+            # registration_data = [
+            #     ("assetCid", asset_cid),
+            #     ("assetSha256", asset_sha256),
+            #     ("assetMimetype", asset_mime_type),
+            #     ("assetTimestampCreated", asset_timestamp_created),
+            #     ("custom", json.dumps(custom)),
+            #     ("nftContractAddress", nft_contract_address),
+            #     ("nftMetadata", json.dumps(nft_metadata)),
+            # ]
 
-            registration_data = [
-                ("assetCid", asset_cid),
-                ("assetSha256", asset_sha256),
-                ("assetMimetype", asset_mime_type),
-                ("assetTimestampCreated", asset_timestamp_created),
-                ("custom", json.dumps(custom)),
-                ("nftContractAddress", nft_contract_address),
-                ("nftMetadata", json.dumps(nft_metadata)),
-            ]
-
-        resp = requests.post(
-            _REGISTER,
-            headers={"Authorization": f"Bearer {config.NUMBERS_API_KEY}"},
-            data=registration_data,
-        )
-
-        if not resp.ok:
-            _logger.error(
-                f"Numbers registration failed: {resp.status_code} {resp.text}"
-            )
-            return None
+            raise NotImplementedError("Don't know how to handle nft contract address")
 
-        data = resp.json()
-        if data.get("response") is None:
-            _logger.warning(
-                "Numbers registration response did not have the 'response' field: %s",
-                resp.text,
-            )
-            return None
-        if data.get("status") != "success":
-            _logger.warning(
-                "Numbers registration didn't have success status: %s", resp.text
+        result = {}
+
+        for chain in chains:
+            if chain == "numbers":
+                server = "https://eo883tj75azolos.m.pipedream.net"
```

Comment:
These URLs need to be passed thru `.env.example`. Avoid having URLs in code impl.